src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -244,8 +254,9 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
+                    _channel.Close();
                     _channel.Dispose();
-                    _connection.Dispose();
+                    _connectionFactory.ReleaseConnection();
```

Comment:
We might not actually release the connection here. Can we rename this method to reflect that? Perhaps `TryReleaseConnection` (we'd need to also return a boolean to indicate whether the connection was released, but we wouldn't need to do anything with this result here).